tests/phpunit/tests/theme/wpThemeJson.php

```diff
@@ -3493,4 +3493,335 @@ function test_get_styles_for_block_with_content_width() {
 		$style_rules = $theme_json->get_styles_for_block( $metadata );
 		$this->assertSame( $expected, $root_rules . $style_rules );
 	}
+
+	/**
+	 * @dataProvider data_generate_spacing_scale_fixtures
+	 *
+	 * @ticket 56467
+	 *
+	 * Tests generating the spacing presets array based on the spacing scale provided.
+	 */
+	function test_set_spacing_sizes( $spacing_scale, $expected_output ) {
+		$theme_json = new WP_Theme_JSON(
+			array(
+				'version'  => 2,
+				'settings' => array(
+					'spacing' => array(
+						'spacingScale' => $spacing_scale,
+					),
+				),
+			)
+		);
+
+		$theme_json->set_spacing_sizes();
+		$this->assertSame( $expected_output, _wp_array_get( $theme_json->get_raw_data(), array( 'settings', 'spacing', 'spacingSizes', 'default' ) ) );
+	}
+
+	/**
+	 * Data provider for spacing scale tests.
+	 *
+	 * @return array
+	 */
+	function data_generate_spacing_scale_fixtures() {
+		return array(
+			'one_step_spacing_scale' => array(
+				'spacingScale'    => array(
+					'operator'   => '+',
+					'increment'  => 1.5,
+					'steps'      => 1,
+					'mediumStep' => 4,
+					'unit'       => 'rem',
+				),
+				'expected_output' => array(
+					array(
+						'name' => 'Medium',
+						'slug' => '50',
+						'size' => '4rem',
+					),
+				),
+			),
+
```

Comment:
The empty lines between datasets can be removed.

Reply:
done